pkg/config/config.go

```diff
@@ -90,24 +101,159 @@ func init() {
 	}
 
 	viper.SetDefault("MessageTemplate", defaultMessageTemplate)
-	viper.SetDefault("Verbose", true)
+	viper.SetDefault("Verbose", false)
 	viper.SetDefault("ListenPort", 8080)
 
 	err = viper.Unmarshal(&AppConfig)
 	if err != nil {
 		panic(err)
 	}
 
-	for _, x := range []*string{
-		&AppConfig.LDAPConfig.Host,
-		&AppConfig.JiraConfig.Host,
-		&AppConfig.SplunkConfig.Host,
-		&AppConfig.SplunkConfig.Token,
-	} {
-		_, err := url.Parse(*x)
-		if err != nil {
-			panic(err)
+	if !AppConfig.Valid() {
+		// If the config is invalid, log the errors and exit right away
+		log.Fatal("FATAL: Configuration invalid - exiting")
+	}
+}
+
+// Valid() wraps the validation functions for the config struct
+// and collects all the errors returned so they can be logged
+// together rather than erroring out on the first one found.
+// This allows the user to fix all the errors at once rather than
+// having to fix one, run the program, fix the next one, etc.
+func (a *Config) Valid() bool {
+	var configErrors []error
+
+	validationFunctions := []func(a *Config) []error{
+		fieldsAreNotNil,
+		hostFieldsAreParsable,
+		passwordOrTokenExistIfUsernameProvided,
+	}
+
+	for _, f := range validationFunctions {
+		configErrors = append(configErrors, f(a)...)
+	}
+
+	if configErrors != nil {
+		log.Print(errors.Join(configErrors...))
+		return false
+	}
+
+	return true
+}
```

Comment:
@mjlshen - I incorporated your changes, and adjusted this a little to simplify more.